code/modules/mob/living/carbon/human/species.dm

```diff
@@ -678,7 +678,8 @@
 				var/obj/item/clothing/glasses/G = H.glasses
 				H.sight |= G.vision_flags
 				H.see_in_dark = G.darkness_view
-				H.see_invisible = G.invis_view
+				if(G.invis_view < H.see_invisible)
```

Comment:
wait... isn't this the wrong way round?

"if the glasses see invisibility is less than the human's see invisibility, override the human's with that of the glasses"